brillig_bytecode/src/opcodes.rs

```diff
@@ -0,0 +1,217 @@
+    Or,  // (|) Bitwise OR
+    Xor, // (^) Bitwise XOR
+    Shl, // (<<) Shift left
+    Shr, // (>>) Shift right
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Comparison {
+    Eq,  //(==) equal
+    Lt,  //(<) field less
+    Lte, //(<=) field less or equal
+}
+
+impl BinaryFieldOp {
+    /// Evaluate a binary operation on two FieldElements and return the result as a FieldElement.
+    pub fn evaluate_field(&self, a: FieldElement, b: FieldElement) -> FieldElement {
+        match self {
+            // Perform addition, subtraction, multiplication, and division based on the BinaryOp variant.
+            BinaryFieldOp::Add => a + b,
+            BinaryFieldOp::Sub => a - b,
+            BinaryFieldOp::Mul => a * b,
+            BinaryFieldOp::Div => a / b,
+            // Perform a comparison between a and b based on the Comparison variant.
+            BinaryFieldOp::Cmp(comparison) => match comparison {
+                Comparison::Eq => ((a == b) as u128).into(),
+                Comparison::Lt => ((a < b) as u128).into(),
+                Comparison::Lte => ((a <= b) as u128).into(),
+            },
+        }
+    }
+}
+
+impl BinaryIntOp {
+    /// Evaluate a binary operation on two unsigned integers (u128) with a given bit size and return the result as a u128.
+    pub fn evaluate_int(&self, a: u128, b: u128, bit_size: u32) -> u128 {
+        let bit_modulo = 1_u128 << bit_size;
+        match self {
+            // Perform addition, subtraction, and multiplication, applying a modulo operation to keep the result within the bit size.
+            BinaryIntOp::Add => (a + b) % bit_modulo,
+            BinaryIntOp::Sub => (a - b) % bit_modulo,
+            BinaryIntOp::Mul => (a * b) % bit_modulo,
+            // Perform unsigned division using the modulo operation on a and b.
+            BinaryIntOp::UnsignedDiv => (a % bit_modulo) / (b % bit_modulo),
+            // Perform signed division by first converting a and b to signed integers and then back to unsigned after the operation.
+            BinaryIntOp::SignedDiv => {
+                to_unsigned(to_signed(a, bit_size) / to_signed(b, bit_size), bit_size)
+            }
+            // Perform a comparison between a and b based on the Comparison variant.
+            BinaryIntOp::Cmp(comparison) => match comparison {
+                Comparison::Eq => (a == b) as u128,
+                Comparison::Lt => (a < b) as u128,
+                Comparison::Lte => (a <= b) as u128,
+            },
+            // Perform bitwise AND, OR, XOR, left shift, and right shift operations, applying a modulo operation to keep the result within the bit size.
+            BinaryIntOp::And => (a & b) % bit_modulo,
+            BinaryIntOp::Or => (a | b) % bit_modulo,
+            BinaryIntOp::Xor => (a ^ b) % bit_modulo,
+            BinaryIntOp::Shl => (a << b) % bit_modulo,
+            BinaryIntOp::Shr => (a >> b) % bit_modulo,
+        }
+    }
+}
+
```

Comment:
There should be a Rust function to convert between unsigned and signed -- In either case, these should be tested in the next refactor

Reply:
Created issue #304